src/main/java/org/sqlite/core/NativeDB.c

```diff
@@ -616,10 +650,6 @@ void change_busy_handler(JNIEnv *env, jobject nativeDB, jobject busyHandler)
         (*env)->GetJavaVM(env, &busyHandlerContext->vm);
 
         busyHandlerContext->obj = (*env)->NewGlobalRef(env, busyHandler);
-        busyHandlerContext->methodId = (*env)->GetMethodID(  env,
-                                                   (*env)->GetObjectClass(env, busyHandlerContext->obj),
-                                                   "callback",
-                                                   "(I)I");
```

Comment:
This did a lookup to `BusyHandler#callback` for every implementation of `BusyHandler` that was passed to this function. It is sufficient to store the reference to the interface method, which will allow calling it on every subclass as well.

The lookup on the subclasses also ment that a jni-config entry would have been needed for every used implementation class which pushes the configuration requirement to the end-user. By storing the interface method id there is no more configuration needed by the user.

Reply:
Nice fix (:+1: